models/solution/route.rb

```diff
@@ -55,6 +55,46 @@ def insert_stop(vrp, stop, index, idle_time = 0)
         shift_route_times(idle_time + stop.activity.duration, index)
       end
 
+      def remove_stop(vrp, index)
+        removed_stop = stops.delete_at(index)
+        previous_stop = stops[0..index - 1].reverse.find{ |n_stop| n_stop.activity.point } if index != 0
+        next_stop = stops[index..-1].find{ |n_stop| n_stop.activity.point } if index != stops.size
+        passed_stop = false
+        non_located_stops = stops[0..index - 1].reverse.select{ |stop|
+          passed_stop = true if stop == removed_stop
```

Comment:
if stop is deleted from stops (`removed_stop = stops.delete_at(index)`)

how can it still be in the `stops` array (`if stop == removed_stop`)

Reply:
it should be previous_stop ;)